lib/whitehall_importer/change_history.rb

```diff
@@ -0,0 +1,41 @@
+module WhitehallImporter
+  class ChangeHistory
+    attr_reader :document_change_history
+
+    def initialize(whitehall_export)
+      @document_change_history = whitehall_document_change_history(whitehall_export).compact
+    end
+
+    def for(edition_number)
+      change_history = document_change_history.select do |edition_change_history|
+        edition_change_history["edition_number"] < edition_number
+      end
+
+      change_history.map { |change| change["change_history_entry"] }.reverse
+    end
+
+  private
+
+    def whitehall_document_change_history(whitehall_export)
+      whitehall_export["editions"].map.with_index do |edition, index|
+        publish_event = EditionHistory.new(edition["revision_history"]).last_state_event("published")
+        edition_number = index + 1
+
+        next if edition["minor_change"] || edition_number == 1
+        next if %w[submitted rejected draft].include?(edition["state"]) && edition["unpublishing"].blank?
```

Comment:
This caught me out a little - it might be better to capture this condition in a method or variable name. I'd guess this is `edition_hasnt_been_published?` ?

Reply:
Good point, have moved it to a method.